docs/source/usage/torch_compile.rst

```diff
@@ -0,0 +1,187 @@
+
+.. code-block:: bash
+
+    $ # Compile the model with level 3, for a general shape
+    $ vllm serve model -O 3
+
+    $ # Compile the model with level 3, for a general shape plus batch size 1
+    $ vllm serve model -O "{'level': 3, 'candidate_compile_sizes': [1]}"
+
+    $ # Compile the model with level 3, for a general shape plus batch sizes 1, 2, 4, 8
+    $ # Note that candidate_compile_sizes from CLI are 1, 2, 3, 4, 5, 6, 7, 8,
+    $ # but the final compiled sizes are 1, 2, 4, 8 because only the ones in cudagraph capture sizes will be compiled.
+    $ vllm serve model -O "{'level': 3, 'candidate_compile_sizes': [$(seq -s, 1 1 8)]}"
+
+When passing the full config from the command line, it is recommended to use double quotes for the outermost quotes and single quotes for the keys of the dictionary. This allows you to use shell commands or environment variables in the value part, as the shell will not expand variables inside single quotes.
+
+When using vLLM from Python, the ``torch.compile`` flags can be specified via the ``compilation_config`` argument of the ``LLM`` class. For example:
+
+.. code-block:: python
+
+    from vllm import LLM
+
+    # Compile the model with level 3, for a general shape
+    llm = LLM(
+        model="model",
+        compilation_config=3,
+    )
+
+    # Compile the model with level 3, for a general shape plus batch size 1
+    llm = LLM(
+        model="model",
+        compilation_config={"level": 3, "candidate_compile_sizes": [1]},
+    )
+
+    # Compile the model with level 3, for a general shape plus batch sizes 1, 2, 4, 8
+    llm = LLM(
+        model="model",
+        compilation_config={"level": 3, "candidate_compile_sizes": list(range(1, 9))},
+    )
+
+Understanding the compilation time
+----------------------------------
+
+Compiling the model to achieve speedups takes time. No matter how long the compilation takes, vLLM ensures that all compilation occurs before serving any requests. No compilation will happen during serving time. To understand the compilation time, you can check the vLLM server logs, e.g., ``init engine (profile, create kv cache, warmup model) took 14.18 seconds``. Compare this with and without compilation to gauge the compilation time.
+
+The breakdown of the compilation time is as follows:
+
+- **Dynamo bytecode transform**: Time taken for TorchDynamo to analyze the Python code. Check the logs for ``Dynamo bytecode transform time: 4.60 s``.
+- **Inductor graph compilation**: Time taken for the inductor to compile the computation graph into Triton kernels. It includes compilation for a general shape and specific shapes. Check the logs for ``Compiling a graph for general shape takes 14.77 s`` and ``Compiling a graph for shape 1 takes 13.52 s``.
+- **Triton kernel compilation**: Time taken for Triton to compile the Triton kernels into GPU kernels. No specific logs are available for this part.
```

Comment:
The inductor graph compilation time is inclusive of the triton kernel compilation time, right? Could you clarify that if so?

Reply:
I don't think so. Some triton kernels might have JIT compilation that is not included in the inductor compilation time.